idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/intentions/implicitcoercion.kt

```diff
@@ -1,34 +1,31 @@
 package arrow.meta.ide.plugins.proofs.intentions
 
-import org.jetbrains.kotlin.idea.debugger.sequence.psi.resolveType
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import com.intellij.codeInspection.ProblemHighlightType
 import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
 import org.jetbrains.kotlin.psi.KtElement
-import org.jetbrains.kotlin.types.KotlinType
 
-// deprecated until #
-/*fun IdeMetaPlugin.makeImplicitCoercionIntention(compilerContext: CompilerContext): ExtensionPhase =
-  addIntention(
-  text = "Make coercion implicit",
-  kClass = KtElement::class.java,
-  isApplicableTo = { ktCall: KtElement, _ ->
-    ktCall.implicitParticipatingTypes()?.let { (subtype, supertype) ->
-      compilerContext.areTypesCoerced(subtype, supertype)
-    } ?: false
-  },
-  applyTo = { ktCall: KtElement, _ ->
-    when (ktCall) {
-      is KtDotQualifiedExpression -> {
-        ktCall.replace(ktCall.receiverExpression)
+fun IdeMetaPlugin.implicitCoercionIntention(compilerContext: CompilerContext): ExtensionPhase =
+  addApplicableInspection(
+    defaultFixText = "Make_coercion_implicit",
+    kClass = KtElement::class.java,
+    enabledByDefault = true,
+    isApplicable = { ktCall: KtElement ->
+      ktCall.implicitParticipatingTypes().any { (subtype, supertype) ->
+        compilerContext.areTypesCoerced(subtype, supertype)
       }
-    }
-  }
-)*/
-
-fun KtElement.implicitParticipatingTypes(): Pair<KotlinType, KotlinType>? =
-  when (this) {
-
-    is KtDotQualifiedExpression ->
-      receiverExpression.resolveType() toOrNull selectorExpression?.resolveType()
-
-    else -> null
-  }
\ No newline at end of file
+    },
+    applyTo = { ktCall: KtElement, _, _ ->
+      when (ktCall) {
+        is KtDotQualifiedExpression -> {
+          ktCall.replace(ktCall.receiverExpression)
+        }
+      }
+    },
+    inspectionText = { "TODO impl" },
+    inspectionHighlightType = { ProblemHighlightType.INFORMATION },
+    groupPath = ArrowPath + arrayOf("Coercion")
```

Comment:
here I suggest creating a variable like `ProofPath = ArrowPath + "Type Proofs"`
```suggestion
    groupPath = ProofPath + arrayOf("Coercion")
```